src/test/java/io/strimzi/kafka/metrics/KafkaMetricsCollectorTest.java

```diff
@@ -100,4 +98,28 @@ private KafkaMetric buildNonNumericMetric(String name, String group) {
                 time);
     }
 
-}
+    @Test
+    public void testCollectNonNumericMetric() {
+        Map<String, String> props = new HashMap<>();
+        props.put(PrometheusMetricsReporterConfig.ALLOWLIST_CONFIG, "kafka_server_group_name.*");
+        PrometheusMetricsReporterConfig config = new PrometheusMetricsReporterConfig(props);
+        KafkaMetricsCollector collector = new KafkaMetricsCollector(config);
+        collector.setPrefix("kafka.server");
+
+        List<Collector.MetricFamilySamples> metrics = collector.collect();
+        assertTrue(metrics.isEmpty());
+
+        // Adding a non-numeric metric converted
+        KafkaMetric nonNumericMetric = buildNonNumericMetric("name", "group");
+        collector.addMetric(nonNumericMetric);
+        metrics = collector.collect();
+        Collector.MetricFamilySamples getMetrics = metrics.get(0);
+
+        assertEquals(1, metrics.size());
```

Comment:
assert the size before you access the first element.

```suggestion
        assertEquals(1, metrics.size());
        Collector.MetricFamilySamples getMetrics = metrics.get(0);
```

Reply:
Moved